lib/Onyx.js

```diff
@@ -1080,6 +1086,11 @@ function merge(key, changes) {
                 // After that we merge the batched changes with the existing value
                 const modifiedData = removeNullObjectValues(applyMerge(existingValue, [batchedChanges]));
 
+                // If the data is null, remove it from cache and store
+                if (_.isNull(modifiedData)) {
+                    return remove(key);
+                }
```

Comment:
I think this can happen sooner. If the `batchedChanges` are `null` then we don't need to concern ourselves with the `existingValue` because we are deleting everything anyway.